lib/module.js

```diff
@@ -48,9 +40,9 @@ function Module(id, parent) {
 }
 module.exports = Module;
 
-Module._cache = {};
-Module._pathCache = {};
-Module._extensions = {};
+Module._cache = Object.create(null);
+Module._pathCache = Object.create(null);
+Module._extensions = Object.create(null);
```

Comment:
So far, this is the only bit that may be concerning. I'm hugely +1 on making these changes, but there's obvious risk here. We'll want to smoke test this thoroughly /cc @nodejs/citgm